### PR TITLE
Error page

### LINKED ISSUE #15: Errors
- [x] Make actual error page (svelte)
- [x] Add way to send the error page as a response
- [x] Store error information in DB for debugging
  - [x] Setup schema and types
  - [x] Add to catches and generate error
  - [x] Add error ID to error page for reference

### PR DESCRIPTION
**Pull request by asportnoy**
Adds error page (/error). Providing an error ID as a hash can be used to view a specific error. Caught errors are stored in the database and ID is returned in the response. Note that if you are signed in when the error is created, you must be signed in to view the error page.

Easiest way to get to the error page is to start signing in but press "Cancel".

OAuth cancelled error
Screenshot on 2022-01-12 at 14 55 22
No error ID/not found/unauthorized
Screenshot on 2022-01-12 at 14 56 29
Error ID found
Screenshot on 2022-01-12 at 14 57 25

Closes #15

### CODE REVIEW

File: frontend/src/pages/error.svelte [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+					return;
+				} catch (e) {}
+			}
+		}
+
+		error = {
+			title: 'Something went wrong',
+			description: 'If the issue persists, please contact us.',
+		};
```
Comment (mswider) [inline frontend/src/pages/error.svelte:38]: Could this be moved to the definition above? The properties here would be overwritten if the request succeeds anyways

**Review by mswider — APPROVED** [review]
This is a nice change and a massive improvement over just plain text for errors.

I don't have any issues, but I feel like you could save space by saving a code for the type of error instead of saving the same data over and over to the DB for errors that don't have any dynamic data. You could then use this code for the type of error to add that data back into the request when the user views the error page. 
You don't have to do any of that, just a thought.

File: frontend/src/pages/error.svelte (hunk 1 above)
Comment (asportnoy) [reply frontend/src/pages/error.svelte:38]: Done

Comment (asportnoy) [reply frontend/src/pages/error.svelte:38]: I just remembered why I did this originally. This way no data shows up until it loads. Undoing now.

**Comment by asportnoy** [comment]
The `ErrorDetail` class has static properties for predefined error data with methods to override parts of it.

**Review by mswider — COMMENTED** [review]
I guess it's already merged but it looks good

**Pull request merged**